src/app/services/log.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { LogLevel } from '../models/log-level.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LogService {
+  private readonly logLevel: LogLevel = new LogLevel();
+
+  info(msg: string): void {
+    this.logWith(this.logLevel.Info, msg);
+  }
+  warn(msg: string): void {
+    this.logWith(this.logLevel.Warn, msg);
+  }
+  error(msg: string): void {
+    this.logWith(this.logLevel.Error, msg);
+  }
+
+  private logWith(level: number, msg: string): void {
```

Comment:
I don't think this method is a really useful abstraction. I would recommend just doing the console logs inside each method above.